### DISCUSSION TITLE
Feed a function to company-idle-delay which generates the idle time according to the buffer's mode.

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [company-mode/company-mode#1234] [Q&A] [answered]
According to the documentation [here](https://github.com/company-mode/company-mode/blob/master/NEWS.md#2019-04-15-0910): `company-idle-delay` now accepts a function which generates the idle time or `nil` indicating no idle completion. So I try the following configuration for `company`:

```emacs-lisp
(use-package company
[...]  
  :init
  (defun my/company-idle-delay () 
                   (LaTeX-mode . (lambda () 
                     (setq-local company-idle-delay 0.8)))) 
[...]
  (setq company-idle-delay #'my/company-idle-delay))
```

But the above code snippet doesn't do the trick at all. Any hints for this problem?

Regards,
HZ

**Comment by yugaego** [comment]
Does the following information help?

`git show ebc4fc` outputs:
> Make company-idle-delay support predicate function
> Sample usage:

```elsip
(setq company-idle-delay (lambda () (if (company-in-string-or-comment) nil 0.3)))
```

**Comment by hongyi-zhao** [comment, marked as answer]
Thank you. The following method solved my problem:
```emacs-lisp
(setq company-idle-delay (lambda () (if (derived-mode-p 'latex-mode) 0.8 0.2)))
```

**Reply by hongyi-zhao** [reply]
But I still don't understand why my original custom function does not work.

**Reply by yugaego** [reply]
Was your function generating "the idle time or nil"?

**Reply by hongyi-zhao** [reply]
I've posted the code snippet in [the initial post](https://github.com/company-mode/company-mode/discussions/1234#discussion-3606177) of this discussion:
```emacs-lisp
  (defun my/company-idle-delay () 
                   (LaTeX-mode . (lambda () 
                     (setq-local company-idle-delay 0.8)))) 
```

**Reply by hongyi-zhao** [reply]
Got it. As the documentation said [here](https://github.com/company-mode/company-mode/blob/master/NEWS.md#2019-04-15-0910): `company-idle-delay` now accepts a function which generates the idle time or `nil` indicating no idle completion. So the function provided to `company-idle-delay` must return idle time or `nil`, which condition is not satisfied by the function I provided above.

**Reply by yugaego** [reply]
Yup.
